src/ctx.cpp

```diff
@@ -2248,20 +2273,22 @@ llvm::Value *FunctionEmitContext::SwitchBoolSize(llvm::Value *value, llvm::Type
     llvm::Value *newBool = value;
     if (g->target->getDataLayout()->getTypeSizeInBits(fromType) >
         g->target->getDataLayout()->getTypeSizeInBits(toType)) {
-        if (name == NULL)
-            name = LLVMGetName(value, "_switchBool");
+        if (name.isTriviallyEmpty()) {
```

Comment:
Should we do that once before the if statement?

Reply:
I changed it. But not to before if statement. 
1. There are conditions where from and to types have same size where we won't go into either the if or else condition
2. If it's outside, a new llvm:Twine will have to be created.

Reply:
Creating Twine is very cheap - that's by design. So if Twine is created and then discarded it should not be expensive.
Anyway, I'm fine with current code.